physics/closed_system.hpp

```diff
@@ -0,0 +1,86 @@
+﻿#pragma once
+
+#include <optional>
+#include <vector>
+
+#include "physics/inertia_tensor.hpp"
+#include "physics/rigid_motion.hpp"
+
+namespace principia {
+namespace physics {
+namespace internal_closed_system {
+
+using geometry::BarycentreCalculator;
+using geometry::Bivector;
+using geometry::Frame;
+using geometry::OrthogonalMap;
+using geometry::SymmetricBilinearForm;
+using quantities::AngularMomentum;
+using quantities::Mass;
+using quantities::MomentOfInertia;
+
+// Computes the instantaneous overall properties of a mechanical system.
+// Effectively an extension of
+// |BarycentreCalculator<DegreesOfFreedom<InertialFrame>, Mass>| that also
+// handles rotational motion.
+// Note: As a consequence of our separate handling of vectors and bivectors, the
```

Comment:
The fact that you have to explain the gory details of the representation is a clear indication that the form is not the right level of abstraction.  This is exactly where `InertiaTensor` (or a simplified version thereof) would come handy.  With the current API there is no abstraction to help diagonalize the tensor to obtain the principal moments and axes.

Reply:
As discussed, this abstraction is `SymmetricBilinearForm`: a symmetric bilinear form over V is only a symmetric bilinear form over V∧V via the anticommutator, and it is obvious in higher dimensions that multiplying a bilinear form over V with an element of V∧V is nonsensical.

I has different eigenvalues over V than {I, _} does over V∧V (but the eigenbases are consistent under ∧).